Zerda is crashing Gmail and some other Apps.

STR:

Open system setting, set Zerda as the default browser to Zerda preview.

Now whenever you try to open Gmail, Gmail crashes.

This definitely sounds like a P0, and it is there for at least 2 days.

Looks like it is related to the blocking custom tab logic:

08-17 12:06:52.567 6083-6083/? E/AndroidRuntime: FATAL EXCEPTION: main
                                                 Process: com.google.android.gm, PID: 6083
                                                 java.lang.RuntimeException: Unable to resume activity {com.google.android.gm/com.google.android.gm.ConversationListActivityGmail}: java.lang.SecurityException: Not allowed to bind to service Intent { act=android.support.customtabs.action.CustomTabsService pkg=org.mozilla.zerda.debug.tyu }
                                                     at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3430)
                                                     at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3470)
                                                     at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2733)
                                                     at android.app.ActivityThread.-wrap12(ActivityThread.java)
                                                     at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1478)
                                                     at android.os.Handler.dispatchMessage(Handler.java:102)
                                                     at android.os.Looper.loop(Looper.java:154)
                                                     at android.app.ActivityThread.main(ActivityThread.java:6121)
                                                     at java.lang.reflect.Method.invoke(Native Method)
                                                     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
                                                     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
                                                  Caused by: java.lang.SecurityException: Not allowed to bind to service Intent { act=android.support.customtabs.action.CustomTabsService pkg=org.mozilla.zerda.debug.tyu }
                                                     at android.app.ContextImpl.bindServiceCommon(ContextImpl.java:1468)
                                                     at android.app.ContextImpl.bindService(ContextImpl.java:1422)
                                                     at android.content.ContextWrapper.bindService(ContextWrapper.java:636)
                                                     at eem.b(SourceFile:329)
                                                     at ikz.b(SourceFile:79)
                                                     at ikz.onActivityResumed(SourceFile:39)
                                                     at android.app.Application.dispatchActivityResumed(Application.java:216)
                                                     at android.app.Activity.onResume(Activity.java:1251)
                                                     at ez.onResume(SourceFile:140)
                                                     at com.android.mail.ui.MailActivity.onResume(SourceFile:105)
                                                     at com.google.android.gm.ui.MailActivityGmail.onResume(SourceFile:238)
                                                     at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1269)
                                                     at android.app.Activity.performResume(Activity.java:6786)
                                                     at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3407)
                                                     at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3470) 
                                                     at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2733) 
                                                     at android.app.ActivityThread.-wrap12(ActivityThread.java) 
                                                     at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1478) 
                                                     at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                     at android.os.Looper.loop(Looper.java:154) 
                                                     at android.app.ActivityThread.main(ActivityThread.java:6121) 
                                                     at java.lang.reflect.Method.invoke(Native Method) 
                                                     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889) 
                                                     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779) 

--- a/app/src/main/AndroidManifest.xml
+++ b/app/src/main/AndroidManifest.xml
@@ -71,13 +71,12 @@
             android:screenOrientation="portrait"
             android:theme="@style/InfoTheme" />
 
+        <!-- Custom Tabs Service is currently not registering the intent-filter before its ready.
+             It should also be set to exported=true if you want to turn it on -->
         <service
             android:name=".customtabs.CustomTabsService"
             android:exported="false"
             tools:ignore="ExportedService">
-            <intent-filter>
-                <action android:name="android.support.customtabs.action.CustomTabsService" />
-            </intent-filter>
         </service>
 
         <meta-data android:name="android.webkit.WebView.MetricsOptOut"

